WIP: auth translation key tidy

darrenvong opened the pull request:
<!---
Please describe the pull request below.
If it closes an issue, make sure to write "closes Couchers-org/couchers#1234"
If there is an issue but it isn't completely closed, still refer to the issue number, eg. "part of Couchers-org/couchers#1234"
--->
Relates to Couchers-org/couchers#4152 for the "auth" feature area. This PR does another sweep to refactor things based on things we recently agreed on (e.g. only importing from `i18n` rather than the module directly), as well as aiming to move some straggler components over to using the JSON file that were missed the first time (looks like there were still quite a few 😢 )

<!---
Checklists - you can remove one that is not applicable (ie. remove backend checklist if you only worked on the web frontend)
If you need help with any of these, please ask :)
--->

**Web frontend checklist**
- [x] Formatted my code with `yarn format && yarn lint --fix`
- [x] There are no warnings from `yarn lint`
- [x] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes


<!---
Remember to request review from couchers-org/web, couchers-org/@backend or an individual.
Once your code is approved, remember to merge it if you have write access
--->

darrenvong committed:
refactor: cleanup auth page constants

darrenvong committed:
refactor: fully i18n jail page

darrenvong committed:
refactor: fully i18n login/signup page

darrenvong committed:
refactor: i18n notification settings...

darrenvong committed:
refactor: fix login tests

darrenvong committed:
refactor: i18n timezone component

Also to stop using the spread pattern and pass props in explicitly.

darrenvong committed:
refactor: add typecheck command

darrenvong committed:
refactor: swap all the things to import from i18n in auth

darrenvong committed:
refactor: fully nuke unnecessary object spread prop passing

darrenvong committed:
refactor: i18n AuthProvider

darrenvong committed:
refactor: final cleanup of translation keys

tsuruby commented on app/web/features/auth/Settings.tsx:
```diff
@@ -39,42 +33,43 @@ export default function Settings() {
-        title={CHANGE_GENDER}
-        content={CHANGE_GENDER_CONTACT}
+        title={t("account_settings_page.gender_section.title")}
+        content={t("account_settings_page.gender_section.explanation")}
```
It is probably not scope of this ticket but I saw that for example in `features/donations` we use "body" instead of "explanation".  And for "jail" it is "description".

darrenvong replied:
That's a good point, maybe we should have some convention on standardising how we name these things... but that can be tricky as it really depends on the context of the component.

For example with this, I'm trying to name all of the sections consistently with "title", "description" and "explanation"... except the gender and birthdate sections don't have a description. I guess the "explanation" could change to something else, so maybe could name it `description2` or something like that.

darrenvong replied:
But I'll be looking at the donations keys next, so I'll change it there if we come to some agreement on what's best :)

tsuruby commented on app/web/features/auth/Settings.tsx:
```diff
@@ -29,6 +22,7 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 export default function Settings() {
+  const { t } = useTranslation("auth");
```
```suggestion
  import { AUTH } from "i18n/namespaces";
  const { t } = useTranslation(AUTH);
```

darrenvong replied:
Ahh nice catch, thanks! 👍

darrenvong committed:
refactor: move timezone and username into settings page object

darrenvong committed:
refactor: remove reference to other translation keys

So it's not "over DRY" and allows other languages max flexibility when
translating

tsuruby reviewed (approved):
lgtm
